habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java

```diff
@@ -0,0 +1,162 @@
+                        key = line.substring(0, line.indexOf(TomlUtils.EQUALS));
+                    }
+
+                    if (key != null) {
+                        key = key.strip();
+
+                        TomlReplacementTuple matchedTuple = replacements.get(key);
+                        if ((matchedTuple != null) && (line.contains(TomlUtils.POETRY_CORE)) && (line.contains(TomlUtils.EQUALS))) {
+                            // update the poetry-core major version upgrade if required.
+                            StringBuilder stringBuilder = new StringBuilder(line);
+                            if(isMajorVersionUpdateRequired){
+                                int majorIndex = line.lastIndexOf(TomlUtils.EQUALS) + 1;
+                                int endMajorIndex = line.indexOf(TomlUtils.DOT);
+                                String poetryCoreReqMajorVer = PoetryUtil.getMajorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT);
+                                stringBuilder = stringBuilder.replace(majorIndex, endMajorIndex, poetryCoreReqMajorVer);
+                                line = stringBuilder.toString();
+                                logger.info("Updating the poetry-core major version.");
+                            }
+
+                            // update the poetry-core minor version if required.
+                            if (isMinorVersionUpdateRequired) {
+                                int minorVersionIndex = line.indexOf(TomlUtils.DOT) + 1;
+                                int endMinorVersionIndex = line.lastIndexOf(TomlUtils.DOT);
+                                String strPoetryCoreReqMinorVersion = PoetryUtil.getMinorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT);
+                                stringBuilder = stringBuilder.replace(minorVersionIndex, endMinorVersionIndex, strPoetryCoreReqMinorVersion);
+                                line = stringBuilder.toString();
+                                logger.info("Updating the Poetry-core major version to {}.", strPoetryCoreReqMinorVersion);
+                            }
+                            logger.info("Updated the Poetry-core version to {} required by Habushu.", PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT);
+                        }
+                    }
+                }
+                fileContent += line + "\n";
+                line = reader.readLine();
+            }
+
+        } catch (IOException e) {
+            throw new HabushuException("Problem reading pyproject.toml while updating version for poetry-core dependencies!", e);
+        }
+
+        try {
+            TomlUtils.writeTomlFile(pyProjectTomlFile, fileContent);
+        } catch (IOException e) {
+            throw new BatonException("Problem while writing dependencies to TomlFile during upgrade of poetry-core version!", e);
+        }
+        return true;
+    }
+
+    private boolean IsPoetrycoreUpgradeRequired(Map.Entry<String, Object> dependency) {
```

Comment:
S: `IsPoetrycoreUpgradeRequired" should be lowercased to "isPoetrycoreUpgradeRequired"